pallets/crowdloan-rewards/src/lib.rs

```diff
@@ -0,0 +1,215 @@
+//! relay chain actually. In this case the genesis config contains information about the
+//! relay chain style contributor address, and the contribution amount. In this case the
+//! contributor is responsible for making a transaction that associates a native ID. The tx
+//! includes a signature by the relay chain idetity over the native identity.
+//!
+//! * **ReadingRelayState**
+//!
+//! The most elegant, but most complex solution would be for the para to read the contributions
+//! directly from the relay state. Blocked by https://github.com/paritytech/cumulus/issues/320 so
+//! I won't persue it further right now. I can't decide whether that would really add security /
+//! trustlessness, or is just a sexy blockchain thing to do. Contributors can always audit the
+//! genesis block and make sure their contribution is in it, so in that sense reading relay state
+//! isn't necessary. But if a single contribution is left out, the rest of the contributors might
+//! not care enough to delay network launch. The little guy might get sensored.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::pallet;
+
+pub use pallet::*;
+
+#[pallet]
+pub mod pallet {
+
+	use frame_support::debug;
+	use frame_support::pallet_prelude::*;
+	use frame_support::traits::Randomness;
+	use frame_support::traits::Vec;
+	use frame_system::pallet_prelude::*;
+	use sp_core::H256;
+	use sp_runtime::Percent;
+
+	/// The Author Filter pallet
+	#[pallet::pallet]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	/// Configuration trait of this pallet.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The overarching event type
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		/// The currency in which the rewards will be paid (probably the parachain native currency)
+		type RewardCurrency: Currency<T::AccountId>;
+		/// The AccountId type contributors used on the relay chain.
+		type RelayChainAccountId; // What trait bounds do I need here? I think concretely we would
+		// be using MultiSigner? Or maybe MultiAccount?
+	}
+
+	type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+	// No hooks
+	//TODO maybe automated payments
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Associate a native identity with a relay chain identity that made a contribution to the
+		/// Crowdloan. Rewards will be paid the the native identity.
+		#[pallet::weight(0)]
+		pub fn associate_native_identity(origin: OriginFor<T>, proof: Vec<u8>) -> DispatchResultWithPostInfo {
+			ensure_signed(origin)?;
+
+			//TODO check the proof:
+			// 1. Is signed by an actual contributor
+			// 2. Signes a valid native identity
+			// 3. The signed identity is the caller
+			//TODO I don't think #3 is actually necessary, and it might complicate things.
```

Comment:
I agree with you, the native identity does not necessarily need to be the caller. There might be a case where someone wants to donate rewards to a native address not in his/her control,